src/tests/ftest/util/data_mover_test_base.py

```diff
@@ -114,23 +129,23 @@ def setUp(self):
         # initialize daos_cmd
         self.daos_cmd = self.get_daos_command()
 
-        # Get the processes for each explicitly
-        # This is needed because both IorTestBase and MdtestBase
-        # define self.processes
-        self.ior_processes = self.params.get(
-            "np", '/run/ior/client_processes/*')
-        self.mdtest_processes = self.params.get(
-            "np", '/run/mdtest/client_processes/*')
-        self.dcp_processes = self.params.get(
-            "np", "/run/dcp/client_processes/*", 1)
-        self.dsync_processes = self.params.get(
-            "np", "/run/dsync/client_processes/*", 1)
-        self.dserialize_processes = self.params.get(
-            "np", "/run/dserialize/client_processes/*", 1)
-        self.ddeserialize_processes = self.params.get(
-            "np", "/run/ddeserialize/client_processes/*", 1)
-
-        self.posix_root = self.params.get("posix_root", "/run/datamover/*", self.tmp)
+        # Get the processes and np for all datamover tools, as well as for individual tools.
+        self.processes = self.params.get("np", '/run/datamover/*', 1)
+        self.ppn = self.params.get("ppn", '/run/datamover/*', 1)
+        self.ior_np = self.params.get("np", '/run/ior/client_processes/*', 1)
+        self.ior_ppn = self.params.get("ppn", '/run/ior/client_processes/*', None)
+        self.mdtest_np = self.params.get("np", '/run/mdtest/client_processes/*', 1)
+        self.mdtest_ppn = self.params.get("ppn", '/run/mdtest/client_processes/*', None)
+        self.dcp_np = self.params.get("np", "/run/dcp/*", self.processes)
+        self.dcp_ppn = self.params.get("ppn", "/run/dcp/*", self.ppn)
+        self.dsync_np = self.params.get("np", "/run/dsync/*", self.processes)
+        self.dsync_ppn = self.params.get("ppn", "/run/dsync/*", self.ppn)
+        self.dserialize_np = self.params.get("np", "/run/dserialize/*", self.processes)
+        self.dserialize_ppn = self.params.get("ppn", "/run/dserialize/*", self.ppn)
+        self.ddeserialize_np = self.params.get("np", "/run/ddeserialize/*", self.processes)
+        self.ddeserialize_ppn = self.params.get("ppn", "/run/ddeserialize/*", self.ppn)
+
+        self.posix_root.update(self.params.get("posix_root", "/run/datamover/*", self.tmp))
```

Comment:
Assign the default with creating the `KeywordParameter` object and then use it's `get_yaml_value()` method to read the test yaml:
```suggestion
        self.posix_root.get_yaml_value("posix_root", self, "/run/datamover/*")
```